pkg/inputs/snmp/snmp.go

```diff
@@ -204,6 +204,20 @@ func runSnmpPolling(ctx context.Context, snmpFile string, jchfChan chan []*kt.JC
 			}
 		}
 
+		// Tweak any per provider tags and match attributes here.
+		for key, tag := range device.UserTags {
+			delete(device.UserTags, key)
```

Comment:
I'm curious about the ordering here. I _think_  this is deleting the existing userTag before we know there's a match from the global `provider_maps`, should that be the case? Does it make sense that there would be userTags set on a device that we don't want to touch? It feels like there's potentially a precedence rule we need to clear up:

```
least specific                           most specific
global        -> provider_tags        -> device user_tags / match_attributes 
```

Reply:
Yar -- you were unfortunately right here. Lots of tests later and I think this does what is lade out above.